### ISSUE TITLE
Adding arguments hence removing comments?

### ISSUE THREAD

**Issue by simon-tiger** [IDMNYU/p5.js-speech#10]
This github issue is referring to pull request #7.

I'm trying to add some arguments to the `start()` function in the `p5.SpeechRec` object. But if the comments say: "this is one 'start' cycle.", I think we should remove them because I added a continuous mode!

As you can see in commit a2a5d389e57591171581fcab95697eb96b6f00a6, there are some comments. Which look like this:
```javascript
// start the speech recognition engine.  this will prompt a
// security dialog in the browser asking for permission to
// use the microphone.  this permission will persist throughout
// this one 'start' cycle.  if you need to recognize speech more
// than once, use continuous mode rather than firing start()
// multiple times in a single script.
```
The comments are right before the `start()` function in the `p5.SpeechRec` object. But the commit adds arguments to this function:
```javascript
p5.SpeechRec.prototype.start = function(continuous, interimResults) {
  if('webkitSpeechRecognition' in window) {
    this.rec.continuous = continuous;
    this.rec.interimResults = interimResults;
    this.rec.start();
  }
}
```
And before, that piece of code looked like this:
```javascript
p5.SpeechRec.prototype.start = function() {
  if('webkitSpeechRecognition' in window) {
    this.rec.continuous = this.continuous;
    this.rec.interimResults = this.interimResults;
    this.rec.start();
  }
}
```
Are the comments "unnecessary" now? In other words, Should we remove them or leave them there?

**Comment by shiffman** [comment]
I think the comments can stay but should likely be rewritten to address the new "default" continuous mode. @rev3rend are you ok with setting continuous as on by default? I think interimResults should be off, but it's rare that someone just wants to get speech recognized once and only once.

**Comment by rev3rend** [comment]
we should think about it, but you'd have to convince me. 

if we turn continuous on by default, that means the microphone is streaming to the google cloud speech recognition service as long as the sketch stays open, no?  to my mind, that's a security / privacy issue.  it's also a bandwidth hog... you should only be streaming speech to the recognition server when you actually want speech recognized.

for a simple interactive sketch (e.g. the 'turtle' in the p5.speech examples), you'd clearly want continuous mode (*and* interimResults for faster, less accurate callbacks), but if you're making, e.g. a game that's voice-driven you only want to be checking for speech when the code is waiting on a user interaction to proceed.  otherwise it's (a) eavesdropping on the room and (b) firing callbacks constantly that you'd have to design around anyway.   you can 'restart' the object anytime you want to reopen the mic for a single round of recognition, which is safer and more efficient.  

i agree that most people are going to *think* they want continuous speech recognition, but by leaving continuous off it has to be an affirmative choice on the part of the developer.  also, when you're teaching p5 you have to at least have that discussion with your students. :)

fwiw, the default behavior of the underlying API is continuous is off.

**Comment by shiffman** [comment]
Ah, thanks for this reply. I wasn't thinking about it this way. @rev3rend you are absolutely right, the privacy / bandwidth considerations should take precedence. This needs to be addressed in the pull request(s) (#7 #8 #9) which adjust the `start()` function to take arguments.

**Comment by shiffman** [comment]
@simon-tiger thanks for your contribution! @rev3rend and I met and went over the library, I am happy to chat with you on slack about the decisions we made. You can close this issue now!
